src/pages/NIHicWebform.js

```diff
@@ -0,0 +1,2795 @@
+    let datasets = this.state.allDatasetNames;
+    // create dataset case
+    if (fp.isEmpty(oldDataset)) {
+      return !fp.contains(name, datasets);
+    }
+    // update dataset case (include old dataset name as valid)
+    else {
+      let updateDatasetName = fp.find(p => p.propertyName === "Dataset Name", this.state.updateDataset.properties).propertyValue;
+
+      return (name === updateDatasetName || !fp.contains(name, datasets));
+    }
+  };
+
+  showDatasetNameErrors(name, showValidationMessages) {
+    if (fp.isEmpty(name)) {
+      return showValidationMessages ? 'form-control required-field-error' : 'form-control';
+    }
+    // if there is a name loaded in because this is an update
+    if (!fp.isEmpty(this.state.updateDataset)) {
+      let updateDatasetName = fp.find(p => p.propertyName === "Dataset Name", this.state.updateDataset.properties).propertyValue;
+      if (name === updateDatasetName) {
+        return 'form-control';
+      }
+      // if the old dataset name has been edited
+      else {
+        return this.validateDatasetName(name) ? 'form-control' : 'form-control required-field-error';
+      }
+    }
+    // if a new dataset name is being edited
+    else {
+      return this.validateDatasetName(name) ? 'form-control' : 'form-control required-field-error';
+    }
+  };
+
+  attestAndSave = (e) => {
+    this.setState( prev => {
+      let allValid = this.validateRequiredFields(prev.datasetData);
+      if (allValid) {
+        prev.showDialogSubmit = true;
+        prev.problemLoadingUpdateDataset = false;
+        prev.showValidationMessages = false;
+      }
+      else {
+        prev.showDialogSubmit = false;
+        prev.problemLoadingUpdateDataset = false;
+        prev.showValidationMessages = true;
+      }
+      return prev;
+    });
+  };
+
+  isValid(value) {
+    let isValid = false;
+    if (value !== '' && value !== null && value !== undefined) {
+      isValid = true;
+    }
+    return isValid;
+  };
+
+  dialogHandlerSubmit = (answer) => (e) => {
+    if (answer === true) {
+      let ontologies = [];
+      for (let ontology of this.state.formData.ontologies) {
+        ontologies.push(ontology.item);
+      }
+      this.setState(prev => {
+        if (ontologies.length > 0) {
+          prev.formData.ontologies = ontologies;
+        }
+        for (let key in prev.datasetData) {
+          if (prev.datasetData[key] === '') {
+            prev.datasetData[key] = undefined;
+          }
+        }
+        return prev;
+      }, () => {
+        let formData = this.state.datasetData;
+        this.setState(prev => {
+          prev.disableOkBtn = true;
+          return prev;
+        });
+
+        if (this.state.showValidationMessages) {
+          this.setState({showDialogSubmit: false});
+        }
+        else {
+          let ds = this.formatFormData(formData);
+          if (fp.isEmpty(this.state.updateDataset)) {
+            DataSet.postDatasetForm(ds).then(resp => {
```

Comment:
We need to remove any functionality that does modification. This whole if block should go away.